clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -993,6 +1011,34 @@ public void recordTopicFetchMetrics(String topic, int bytes, int records) {
             }
             recordsFetched.record(records);
         }
+
+        public void recordPartitionFetchLag(TopicPartition tp, long lag) {
+            String name = tp + ".records-lag";
+            Sensor recordsLag = this.metrics.getSensor(name);
+            if (recordsLag == null) {
+                recordsLag = this.metrics.sensor(name);
+                recordsLag.add(this.metrics.metricName(tp + ".records-lag", this.metricGrpName, "The latest lag of the partition"),
```

Comment:
It was mainly for readability. We have the metric name of ".records-lag" (which is the same as the sensor name), ".records-lag-max", ".records-lag-avg". But it is probably OK to just use the constructed string here.